### DISCUSSION TITLE
How to access full profile object on the client?

### DISCUSSION THREAD

**Discussion by victorlmneves** [nextauthjs/next-auth#10945] [Help]
Hi

I'm testing the library and after login on to the client I only have access to the 3 props from the profile object and not to the full profile object.
Is there any reason for the user object to only return the name, email, and image and not the full profile object?

![image](https://github.com/nextauthjs/next-auth/assets/17341789/162eadb3-927f-4515-bd81-2a098964b76a)

Thanks

**Comment by k-taro56** [comment]
Hi,

For security.
https://authjs.dev/reference/core#session.

If you need more help, please describe more.
Thanks.

**Reply by victorlmneves** [reply]
thanks @k-taro56 

then, what's the best way to access it on the client?
I need that because I have a couple of requests triggered on the client that require me to access some of that data to send them to the request header

Thanks

**Reply by k-taro56** [reply]
There are several ways to do this. It depends on whether you use `adapter` (database).

It is difficult to advise without _**MORE DETAILS**_. In particular, it is essential that you tell us about the following

- What kind of data do you need?
- How will you use it? (React server component, Middleware, Route handlers or Client side rendering) The following example may be helpful:
  https://next-auth-example.vercel.app/
